src/_P021_Level.ino

```diff
@@ -60,70 +159,171 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
+        addFormNote(msg);
+      }
+      #endif // ifdef PLUGIN_021_DEBUG
+
+      addRowLabel(F("Input Task"));
+      addTaskSelect(F(P021_GUID_CHECK_TASK), P021_CHECK_TASK);
 
       if (validTaskIndex(P021_CHECK_TASK)) {
-        addRowLabel(F("Check Value"));
-        addTaskValueSelect(F("pvalue"), P021_CHECK_VALUE, P021_CHECK_TASK);
+        addRowLabel(F("Input Value"));
+        addTaskValueSelect(F(P021_GUID_CHECK_VALUE), P021_CHECK_VALUE, P021_CHECK_TASK);
       }
 
-      addFormTextBox(F("Set Level"),  F("psetvalue"), toString(P021_TRIGGER_LEVEL),      8);
+      addFormTextBox(F("Setpoint"),   F(P021_GUID_SETPOINT),   toString(P021_SETPOINT),   8);
 
-      addFormTextBox(F("Hysteresis"), F("physt"),     toString(P021_TRIGGER_HYSTERESIS), 8);
+      addFormTextBox(F("Hysteresis"), F(P021_GUID_HYSTERESIS), toString(P021_HYSTERESIS), 8);
 
-      addFormCheckBox(F("Invert Output"),
-                      F("inv"),
-                      P021_INVERT_OUTPUT == 1);
+      addFormCheckBox(F("Invert Output"), F(P021_GUID_INV_OUTPUT), P021_getConfigBit(P021_FLAGS, P021_INV_OUTPUT));
 
       // inverted flag!
-      addFormCheckBox(F("Save 'Set Level'/'Hysteresis' after change via <pre>config</pre> command"),
-                      F("psave_always"),
+      addFormCheckBox(F("Save 'Setpoint'/'Hysteresis' after change via <pre>config</pre> command"),
+                      F(P021_GUID_DONT_ALWAYS_SAVE),
                       P021_DONT_ALWAYS_SAVE == 0);
-      // # ifndef BUILD_NO_DEBUG
+
       addFormNote(F("Saving settings too often can wear out the flash chip on your ESP!"));
-      // # endif // ifndef BUILD_NO_DEBUG
 
-      addFormNumericBox(F("Auto-save interval"), F("pautosave"), P021_AUTOSAVE_TIMER / 60, 0, 1440); // Present in minutes
+      addFormNumericBox(F("Auto-save interval"), F(P021_GUID_AUTOSAVE_TIMER), P021_AUTOSAVE_TIMER / 60, 0, 1440); // Present in minutes
       addUnit(F("minutes"));
-      // # ifndef BUILD_NO_DEBUG
+
       addFormNote(F("Interval to check if settings are changed via <pre>config</pre> command and saves that. Max. 24h, 0 = Off"));
-      // # endif // ifndef BUILD_NO_DEBUG
+
+      // Settings extension for new operation modes. Will reload the page.
+      addFormSelector_YesNo(F("Extended functionality"), F(P021_GUID_EXT_FUNCT), P021_getConfigBit(P021_FLAGS, P021_EXT_FUNCT), true);
+
+      if (P021_getConfigBit(P021_FLAGS, P021_EXT_FUNCT))
+      {
+        // FormSelector with all operation mode options
+        String options[P021_OPMODE_SIZE]      = { F("Classic"), F("Off"), F("Standby"), F("On"), F("Local"), F("Remote") };
+        int    optionValues[P021_OPMODE_SIZE] =
+        { P021_OPMODE_CLASSIC, P021_OPMODE_OFF, P021_OPMODE_STANDBY, P021_OPMODE_ON, P021_OPMODE_TEMP, P021_OPMODE_REMOTE };
+        addFormSelector(F("Control mode"), F(P021_GUID_OPMODE), P021_OPMODE_SIZE, options, optionValues, P021_OPMODE);
```

Comment:
- Using String for these arrays isn't very efficient, `const __FlashStringHelper * options[] = { F("Classic"), F("Off"), F("Standby"), F("On"), F("Local"), F("Remote") };` is the common way of doing that.
- No need to use a length initializer for the arrays isn't needed. And we have a macro `NR_ELEMENTS(optionValues)` that will calculate the actual length at compile-time for use as the `addFormSelect` length argument.
- Using `const int optionValues[] =...` instead of just `int` allows the compiler to apply a few more optimizations.

Reply:
@TD-er: Agreed there might be use cases for smaller delays. However it is also not user friendly to enter a few days in seconds. With a value of 4 days I have to enter 345600 seconds... As long as the unit is clearly mentioned I don't expect much confusion. I am thinking about a compromise here.

Reply:
Redesigned the way how to handle timers. Now timing parameters are stored in seconds. Two modes of presentation: standard, long timers. Standard mode will show seconds. Long timers mode will show minutes and hours with scaling in WEBFORM_LOAD and WEBFORM_SAVE. Control algorithm will scale seconds to millis() domain.